src/Notes/WC_Admin_Notes_Test_Checkout.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * WooCommerce Admin Test Checkout.
+ *
+ * Adds a note to remind the user to test their store checkout.
+ *
+ * @package WooCommerce Admin
+ */
+
+namespace Automattic\WooCommerce\Admin\Notes;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Notes_Test_Checkout
+ */
+class WC_Admin_Notes_Test_Checkout {
+	/**
+	 * Note traits.
+	 */
+	use NoteTraits;
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-admin-test-checkout';
+
+	const PRODUCT_ADDED_OPTION_NAME = 'wc_admin_note_test_checkout_product_added';
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_action( 'product_page_product_importer', array( $this, 'run_on_product_importer_done' ) );
+		add_action( 'transition_post_status', array( $this, 'run_on_transition_post_status_done' ), 10, 3 );
+	}
+
+	/**
+	 * Runs on product importer steps.
+	 */
+	public function run_on_product_importer_done() {
+		// We're only interested in when the importer completes.
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+		if ( ! isset( $_REQUEST['step'] ) ) {
+			return;
+		}
+		if ( 'done' !== $_REQUEST['step'] ) {
+			return;
+		}
+		// phpcs:enable
+
+		$this->update_product_added_and_possibly_add_note();
+	}
+
+	/**
+	 * Runs when a post status transitions, but we're only interested if it is
+	 * a product being published.
+	 *
+	 * @param string $new_status The new status.
+	 * @param string $old_status The old status.
+	 * @param Post   $post       The post.
+	 */
+	public function run_on_transition_post_status_done( $new_status, $old_status, $post ) {
+		if (
+			'product' !== $post->post_type ||
+			'publish' !== $new_status
+		) {
+			return;
+		}
+
+		$this->update_product_added_and_possibly_add_note();
+	}
+
+	/**
+	 * Updates "wc_admin_note_test_checkout_product_added" in the DB and calls "possibly_add_note" method.
+	 */
+	public function update_product_added_and_possibly_add_note() {
+		update_option( self::PRODUCT_ADDED_OPTION_NAME, true );
+		self::possibly_add_note();
+	}
+
+	/**
+	 * Get the note.
+	 */
+	public static function get_note() {
+		// Make sure a product was published or imported.
+		$product_added = get_option(
+			self::PRODUCT_ADDED_OPTION_NAME
+		);
```

Comment:
Rather than hook into the importer and post status transition hooks, what do you think about checking the status of the products step in the task list instead? Seems like this could be simpler if we were just checking for the products and payments steps to be complete.

Reply:
Really like this idea Jeff!

Reply:
Hi @jeffstieler, thanks for your review.
Diving a little bit into this part of the code, I realized that the check (that later will be used to [set the products state](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Features/OnboardingTasks.php#L111) in the [task list](https://github.com/woocommerce/woocommerce-admin/blob/main/client/task-list/tasks.js#L137)) is calculated in the backend in the method `check_task_completion ` and it's [asking for products with state published or draft](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Features/OnboardingTasks.php#L181). From what I understand, this could be a problem since we need to show the note after the product publication, but the list item could be set as completed without a published product.
Please correct me if I misunderstood your suggestion or I'm forgetting something.

Reply:
Hmm.. looks like that wasn't the original intention and was updated later: https://github.com/woocommerce/woocommerce-admin/pull/2667#discussion_r305182796

Perhaps we should look at only marking the step complete if there are published products? cc: @timmyc

Reply:
Yeah I concur there, but cc @pmcpinto to verify for sure. The product step right now will get marked as done even when a product is in the draft state. But I definitely like tying this note to that step being completed because it will eliminate this note possibly being shown on an established site.

Reply:
Thank you @jeffstieler and @timmyc, I removed the verification of the products with state `draft` in the commit 14d5df2 and it looks like everything is working well.

I was thinking about when this note should be shown. If the user doesn't set a payment method when the `Add my products` task list is completed, the note won't be shown. Also, we don't save a timestamp when we update the products' task list, which means that we don't know when we completed this task list item. 

The note reminds the user `Don't forget to test your checkout`, so it looks like we should show the note if the user adds their first product and has the other conditions we need. For instance, a user could have everything set but no products for a month. Should we show the note if they suddenly add a product?

If the answer is yes, I think that we should verify the published date of the first product, and if it's within the last 30 (or 15, or 10) minutes and fulfills the other conditions, I would trigger the note.
Does this make sense?